lib/FredyRuntime.js

```diff
@@ -26,7 +26,7 @@ class FredyRuntime {
       //modify the url to make sure search order is correctly set
       Promise.resolve(urlModifier(this._providerConfig.url, this._providerConfig.sortByDateParam))
         //scraping the site and try finding new listings
-        .then(this._getListings.bind(this))
+        .then(this._providerConfig.getListings?.bind(this) ?? this._getListings.bind(this))
```

Comment:
it is ok for now,  but needs refinement. Something I wanted to tackle anyway.

Reply:
Sounds good. Out of curiosity, what did you have in mind? Maybe moving to a prototype design? Let providers extend a common prototype and overwrite the given methods if necessary?

Reply:
I'm not yet fully sure what the solution will be, but I was thinking about something similar than what you did..